self-core-impl/src/main/java/com/selfxdsd/core/tasks/ContributorTasks.java

```diff
@@ -0,0 +1,111 @@
+package com.selfxdsd.core.tasks;
+
+import com.selfxdsd.api.Issue;
+import com.selfxdsd.api.Task;
+import com.selfxdsd.api.Tasks;
+import com.selfxdsd.api.storage.Storage;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * Active tasks of a Contributor. This class <b>just represents</b>
+ * the tasks. The actual filtering has to be done in an upper layer.
+ *
+ * @author criske
+ * @version $Id$
+ * @since 0.0.1
+ */
+public final class ContributorTasks implements Tasks {
+
+    /**
+     * Contributor's user name.
+     */
+    private final String username;
+
+    /**
+     * Contributor's provider.
+     */
+    private final String provider;
+
+    /**
+     * The contributor's tasks.
+     */
+    private final List<Task> tasks;
+
+    /**
+     * Self storage, to save new contracts.
+     */
+    private final Storage storage;
+
+    /**
+     * Constructor.
+     *
+     * @param username Contributor's user name.
+     * @param provider Contributor's provider.
+     * @param tasks Contributor's tasks.
+     * @param storage Self's storage, to save new tasks.
+     */
+    public ContributorTasks(final String username,
+                            final String provider,
+                            final List<Task> tasks,
+                            final Storage storage) {
+        this.username = username;
+        this.provider = provider;
+        this.tasks = new ArrayList<>(tasks);
+        this.storage = storage;
+    }
+
+    @Override
+    public Task getById(final String issueId,
+                        final String repoFullName,
+                        final String provider) {
+        return this.storage.tasks().getById(issueId, repoFullName, provider);
+    }
+
+    @Override
+    public Task register(final Issue issue) {
+        if (!this.provider.equals(issue.provider())) {
+            throw new IllegalArgumentException(
+                "The given Issue does not belong to the Provider "
+                    + this.provider + "."
+            );
+        }
+        Task task = this.storage.tasks().register(issue);
+        tasks.add(task);
+        return task;
```

Comment:
@criske I think you should leave this method unimplemented for now (throw UOE) because it should be a little more complex: first, check if the Issue and Contributor belong to the same Project, then add the task in Storage and also **Assign** it to the Contributor (this part is not yet implemented at all).

And it gets a little more complicated because the Task in the DB is actually linked to the Contract (based on role), not to the Contributor. So the link between Task and Contributor is indirect.